### DISCUSSION TITLE
when auth_service: session_recording set to proxy, enhanced recording is invalid

### DISCUSSION THREAD

**Discussion by ssikiki** [gravitational/teleport#6470] [Q&A] [answered]
I have a few nodes, some of them enabled ssh_service and enhanced_recording, some of them just running sshd
so if I want to enable session recording for all nodes, I need to set auth_service session_recording to proxy according to https://goteleport.com/docs/architecture/proxy/#recording-proxy-mode
But I found when session_recording is proxy, enhanced_recording is invalid for the nodes running teleport ssh_service

Is there a way to achieve this requirement，the nodes running teleport ssh_service enable enhanced_recording， while the nodes running sshd can record session ?

**Comment by webvictim** [comment, marked as answer]
Yes, this is a niche case where you cannot satisfy both requirements.

My advice would be to run a separate Teleport cluster with `session_recording: proxy` for your `sshd` nodes, then join that to your main Teleport cluster (with `session_recording: node`) using Teleport's [trusted cluster functionality](https://goteleport.com/docs/trustedclusters/#trusted-clusters). This should allow you to have both sets of nodes working as you wish.
